### ISSUE TITLE
COPY after USER does not yield expected behaviour

### ISSUE THREAD

**Issue by henriklynggaard** [docker/docker#22749]
**Output of `docker version`:**
 Version:      1.11.1
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   5604cbe
 Built:        Tue Apr 26 23:30:23 2016
 OS/Arch:      linux/amd64

**Steps to reproduce the issue:**
WRITE docker file with
USER myuser
COPY . /opt

**Describe the results you received:**
Copied content is owned by root

**Describe the results you expected:**
Copied content is owned by myuser

**Additional information you deem important (e.g. issue happens only occasionally):**
I know there has been several prior issues on same or similar which all appear to have been closed without a fix but that is a mistake in my opinion. 

I understand the need to cater for some level of backward compibility , but to cater for people exploiting bugs as opposed to actually fixing the bugs is going too far.

**Comment by thaJeztah** [comment]
Thanks for reporting, but (as you mentioned), this is a duplicate of https://github.com/docker/docker/issues/6119, and breaking existing Dockerfiles, unfortunately, is not something we can take lightly.

**Comment by runcom** [comment]
should we close this? given the linked issues (and PR) all got closed over time..

**Comment by unclejack** [comment]
I agree with @runcom and @thaJeztah. This is something which is an issue and I've also run into this, but we really can't break existing Dockerfiles.

We need to discuss this and figure out how to implement this so that we make it possible to maintain compatibility with existing Dockerfiles.

I'll close this now as this requires a long discussion before we can move forward and implement a solution for all instructions (ADD, COPY and so on). In my opinion, simply letting USER control the owner of the added files isn't enough because this would force us to sprinkle `USER foo` and `USER bar` to switch from one user to another. The best solution for now will most likely be the squashing of images.

Please feel free to comment.
